hal/device_uid_api.h

```diff
@@ -0,0 +1,73 @@
+/** \addtogroup hal */
+/** @{*/
+/* mbed Microcontroller Library
+ * Copyright (c) 2017 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef MBED_DEVICE_UID_API_H
+#define MBED_DEVICE_UID_API_H
+
+#include <stddef.h>
+#include "device.h"
+
+#if DEVICE_DEVICEUID
+
+#ifndef MBED_DEVICEUID_SIZE
+#error "DEVICE UID Vendor implementation must define macro MBED_DEVICEUID_SIZE with the uid size in bytes!"
+#endif
+
+#ifndef MBED_DEVICEUID_STR_SIZE_MAX
+#warning "DEVICE UID max vendor string length not defined! device_uid_get_str() HAL interface is disabled!"
```

Comment:
Is this warning for having it defined even with value 0 to disable this warning? Just trying to understand this warning (one day we might have warning = error).

Reply:
I can change this to check for not defined or defined with value 0